libs/community/langchain_community/embeddings/textembed.py

```diff
@@ -0,0 +1,390 @@
+            url=f"{self.host}/embedding",
+            headers={
+                "accept": "application/json",
+                "content-type": "application/json",
+                "Authorization": f"Bearer {self.api_key}",
+            },
+            json=dict(
+                input=texts,
+                model=model,
+            ),
+        )
+
+    def _sync_request_embed(
+        self, model: str, batch_texts: List[str]
+    ) -> List[List[float]]:
+        """
+        Sends a synchronous request to the embedding endpoint.
+
+        Args:
+            model (str): The model to use for embedding.
+            batch_texts (List[str]): Batch of texts to embed.
+
+        Returns:
+            List[List[float]]: List of embeddings for the batch.
+
+        Raises:
+            Exception: If the response status is not 200.
+        """
+        response = requests.post(
+            **self._kwargs_post_request(model=model, texts=batch_texts)
+        )
+        if response.status_code != 200:
+            raise Exception(
+                f"extEmbed responded with an unexpected status message "
+                f"{response.status_code}: {response.text}"
+            )
+        return [e["embedding"] for e in response.json()["data"]]
+
+    def embed(self, model: str, texts: List[str]) -> List[List[float]]:
+        """
+        Embeds a list of texts synchronously.
+
+        Args:
+            model (str): The model to use for embedding.
+            texts (List[str]): List of texts to embed.
+
+        Returns:
+            List[List[float]]: List of embeddings for the texts.
+        """
+        perm_texts, unpermute_func = self._permute(texts)
+        perm_texts_batched = self._batch(perm_texts)
+
+        # Request
+        map_args = (
+            self._sync_request_embed,
+            [model] * len(perm_texts_batched),
+            perm_texts_batched,
+        )
+        if len(perm_texts_batched) == 1:
+            embeddings_batch_perm = list(map(*map_args))
+        else:
+            with ThreadPoolExecutor(32) as p:
+                embeddings_batch_perm = list(p.map(*map_args))
+
+        embeddings_perm = self._unbatch(embeddings_batch_perm)
+        embeddings = unpermute_func(embeddings_perm)
+        return embeddings
+
+    async def _async_request(
+        self, session: aiohttp.ClientSession, kwargs: Dict[str, Any]
+    ) -> List[List[float]]:
+        """
+        Sends an asynchronous request to the embedding endpoint.
+
+        Args:
+            session (aiohttp.ClientSession): The aiohttp session for the request.
+            kwargs (Dict[str, Any]): Dictionary of POST request parameters.
+
+        Returns:
+            List[List[float]]: List of embeddings for the request.
+
+        Raises:
+            Exception: If the response status is not 200.
+        """
+        async with session.post(**kwargs) as response:
+            if response.status != 200:
+                raise Exception(
```

Comment:
`response.raise_for_status()` potentially? instead of changing exception class to general Exception?

Reply:
Thank you for the suggestion! I agree that using response.raise_for_status() is a cleaner and more specific way to handle HTTP errors.

Reply:
Up to you, but this makes it difficult to retry on failure based on status code since the status code information is lost.